internal/service/team/data_source_team.go

```diff
@@ -77,21 +80,26 @@ func dataSourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	}
 
 	if err := d.Set("team_id", team.GetId()); err != nil {
-		return diag.FromErr(fmt.Errorf(errorTeamSetting, "name", d.Id(), err))
+		return diag.FromErr(fmt.Errorf(errorTeamSetting, "team_id", d.Id(), err))
 	}
 
 	if err := d.Set("name", team.GetName()); err != nil {
 		return diag.FromErr(fmt.Errorf(errorTeamSetting, "name", d.Id(), err))
 	}
 
-	users, _, err := connV2.TeamsApi.ListTeamUsers(ctx, orgID, team.GetId()).Execute()
+	teamUsers, err := dsschema.AllPages(ctx, func(ctx context.Context, pageNum int) (dsschema.PaginateResponse[admin.CloudAppUser], *http.Response, error) {
```

Comment:
I would refactor the duplicate logic to a standalone function
Maybe something like:

func readUsernames(api TeamsApi, orgID, teamID) ([]string, error)

Reply:
added listAllTeamUsers()